WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -234,11 +234,17 @@ class WeeklyRoundupBackgroundTask: BackgroundTask {
     /// Just a convenience method to know then this task is run, what run date to use as "current".
     ///
     private func currentRunPeriodEndDate() -> Date {
-        Calendar.current.nextDate(
+        let runDate = Calendar.current.nextDate(
             after: Date(),
             matching: runDateComponents,
             matchingPolicy: .nextTime,
             direction: .backward) ?? Date()
+
+        // The run date is when the task is scheduled to run, but the period end date is actually
+        // the previous day at 24:59:59.
+        let periodEndDate = Calendar.current.date(bySettingHour: 0, minute: 0, second: 0, of: runDate)!.addingTimeInterval(TimeInterval.init(-1))
```

Comment:
@diegoreymendez For my own curiosity, I'm wondering how that kind of logic works with timezones?

I don't know the internals of how the Weekly Roundup is implemented, but I was suspecting that the Weekly Roundup stats were computed server-side – where the server has its own notion of "day change" and "previous day" depending on the timezone it's set on – while when the iOS notification is sent (by the server, I assume, not a LocalNotification?) to show those stats on the user's home screen, the timezone the user's iPhone is set to might be different, and thus "previous day" and that `periodEndDate` might mean something else, with a couple of hours differences from what it meant for the server?

Reply:
(Also, nit: the time `24:59:59` doesn't exist, you meant `23:59:59` in that comment 😸 )